packages/jana2/package.py

```diff
@@ -58,14 +58,18 @@ class Jana2(CMakePackage):
 
     variant("root", default=False, description="Use ROOT for janarate.")
     variant("zmq", default=False, description="Use zeroMQ for janacontrol.")
+    variant("podio", default=False, description="Use Podio.")
 
     depends_on("cmake@3.16:", type="build")
     depends_on("cppzmq", when="+zmq")
     depends_on("root", when="+root")
+    depends_on("podio", when="+podio")
```

Comment:
Is there a minimum version?
```suggestion
    depends_on("podio@0.16:", when="+podio")
```

Reply:
Yes, but...
There's a couple of not-strictly-necessary changes that I want to PR to PODIO. This means that JANA releases are going to be tied to precise PODIO releases for a little while. So this coarse `depends_on` relation gives the user a false impression about which versions are actually compatible.

Reply:
I'm guessing it isn't kosher to go in and change this value in the future? Since the spack recipe is supposed to describe the whole history of version compatibility and not just the present?

Reply:
Changing values retroactively is indeed to be avoided. But you can do things like:
```suggestion
    depends_on("podio@0.16:", when="@2.0.9: +podio")
    depends_on("podio@0.16.2:0.16", when="@2.1.0 +podio")
    depends_on("podio@0.17:", when="@2.1.1: +podio")
```
This means just adding lines (not modifying past lines). The most restrictive conditions take effect (you might get a kick out of the underlying translation into ASP rule set).

Though typically we don't go and close the ranges, so once podio 0.17 is out, and when by then jana2 2.1.1 is out we don't go in and check that jana2 2.0.9 still builds with the latest 0.17. 

And I've done even uglier things (https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-awkward/package.py#L52-L63).